### DISCUSSION TITLE
Showing "Download Failed" on Hotstar

### DISCUSSION THREAD

**Discussion by sp8996** [aclap-dev/video-downloadhelper#1461] [Help and Q&A]
### Your question

Hi, I'm trying to download this tv-series and it's always resulting in failure of download. Even though the previous episodes of the same series downloaded just fine. Please try to fix this.
Link: https://www.hotstar.com/in/shows/mahabharat-bangla/1493/two-boys-are-born/1000034115/watch

### Addon & CoApp version and details

version: 9.1.0.41
target: mozilla
channel: beta
lang: en-US
coapp: {"found":true,"path":"/home/home/.local/share/vdhcoapp/vdhcoapp","version":"2.0.19","new_version":false}
license: {"unneeded":true}
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:128.0) Gecko/20100101 Firefox/128.0
{
    "http_media_download_strategy": "coapp",
    "used_history_button": true,
    "concurrent_downloads_max": 20,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_legacy_ui": false,
    "use_sidebar": false,
    "open_count_store": 7,
    "successfull_dl": 64,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": [
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_8879032850573774"
                    },
                    {
                        "download_result": {
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/home/home/dwhelper/Watch Mahabharat Bangla S1 Episode 1 on Disney Hotstar.mp4"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "Watch Mahabharat Bangla S1 Episode 1 on Disney Hotstar.mp4"
                            },
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/home/home/dwhelper"
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.hotstar.com/in/shows/mahabharat-bangla/1493/shantanu-accepts-bhishma-as-his-son/1000034111/watch"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1731580036995
                        }
                    }
                ]
            },
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_1070746679501398"
                    },
                    {
                        "download_result": {
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/home/home/dwhelper/Watch Mahabharat Bangla S1 Episode 2 on Disney Hotstar.mp4"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "Watch Mahabharat Bangla S1 Episode 2 on Disney Hotstar.mp4"
                            },
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/home/home/dwhelper"
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.hotstar.com/in/shows/mahabharat-bangla/1493/bhishma-saves-vichitravirya/1000034112/watch?episodeNumber=1&seasonId=1494"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1731580329165
                        }
                    }
                ]
            },
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_2915452129632379"
                    },
                    {
                        "download_result": {
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/home/home/dwhelper/Watch Mahabharat Bangla S1 Episode 3 on Disney Hotstar.mp4"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "Watch Mahabharat Bangla S1 Episode 3 on Disney Hotstar.mp4"
                            },
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/home/home/dwhelper"
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.hotstar.com/in/shows/mahabharat-bangla/1493/bhishma-defeats-king-salva/1000034113/watch?episodeNumber=2&seasonId=1494"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1731582309369
                        }
                    }
                ]
            },
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_2942402785446379"
                    },
                    {
                        "download_result": {
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/home/home/dwhelper/Watch Mahabharat Bangla S1 Episode 4 on Disney Hotstar.mp4"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "Watch Mahabharat Bangla S1 Episode 4 on Disney Hotstar.mp4"
                            },
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/home/home/dwhelper"
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.hotstar.com/in/shows/mahabharat-bangla/1493/amba-orders-bhishma-to-marry-her/1000034114/watch?episodeNumber=3&seasonId=1494"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1731582324727
                        }
                    }
                ]
            }
        ]
    },
    "record_download_history": true,
    "history_limit_in_days": 1
}

**Comment by Mattmjs** [comment]
Do you see this in Firefox, if you do it's DRM protected. If it's not showing that then more information would be needed.

![DRM icon](https://github.com/user-attachments/assets/e0152fa5-bea4-4900-9f23-5c24de79a4ec)

**Reply by sp8996** [reply]
Hi, just checked, the video also plays with DRM disabled in Firefox. But still VDH won't download.

**Comment by sp8996** [comment]
Yes this icon is showing but the previous episodes can be downloaded also and the same "DRM icon" is showing for those links also. And if I copy the m3u8 file link (of the same episode that I was unable to download), it can be downloaded using yt-dlp.

**Comment by sp8996** [comment]
This is the error message:
```
[in#1 @ 0x1aa46840] Error opening input: Server returned 403 Forbidden (access denied)
Error opening input file https://abgpntkaaaaaaaamaaaaaaaaaaaaa.hses7-vod-cf-ace.cdn.hotstar.com/videos/hotstarint/1000034115/v3/1652421654544/651e1040045f6dee8363ad81f53de239/audio_aac_1652421654544/und/mp4a/index.m3u8.
Error opening input files: Server returned 403 Forbidden (access denied)
```
And for the same m3u8 link:
```
home@debian:~/Downloads/Mahabharat$ ./yt-dlp_linux --cookies-from-browser firefox https://abgpntkaaaaaaaamaaaaaaaaaaaaa.hses7-vod-cf-ace.cdn.hotstar.com/videos/hotstarint/1000034115/v3/1652421654544/651e1040045f6dee8363ad81f53de239/video_fhd_h264_sdr_1652421654544/avc1/5/index.m3u8 -o output.mp4
[generic] Extracting URL: https://abgpntkaaaaaaaamaaaaaaaaaaaaa.hses7-vod-cf-ace.cdn.hotstar.com/videos/hotstarint/10000341...44/avc1/5/index.m3u8
[generic] index: Downloading webpage
Extracting cookies from firefox
Extracted 2729 cookies from firefox
[generic] index: Downloading m3u8 information
[generic] index: Checking m3u8 live status
[info] index: Downloading 1 format(s): 0
[hlsnative] Downloading m3u8 manifest
WARNING: Live HLS streams are not supported by the native downloader. If this is a livestream, please add "--downloader ffmpeg --hls-use-mpegts" to your command
[hlsnative] Total fragments: 307
[download] Destination: output.mp4
[download]  13.7% of ~ 644.73MiB at    6.63MiB/s ETA 01:25 (frag 42/307)
```

**Reply by Mattmjs** [reply]
I tried the m3u8 url in both yt-dlp and ffmpeg while using a vpn connection to India but both still returned a 403 error.
Did you get the 403 error in  ffmpeg or yt-dlp ? VDH uses ffmpeg to download the file and it may well be that it fails on the
403 error message. In that case I'm not sure there is any way around that.

**Comment by sp8996** [comment]
I got error 403 on VDH (old legacy UI as no error logs was there in the new UI). But no error on yt-dlp. 
Did you use browser cookies? Using browser cookies won't result in error 403 I think.

**Reply by Mattmjs** [reply]
No, I'm not a member of the hotstar site. It's requires signing up and paying a subscription to access it.

**Reply by sp8996** [reply]
if you need cookies for testing, i can provide you that.

**Reply by Mattmjs** [reply]
Won't be needed, you've found that VDH
suffers from the 403 error but yt-dlp doesn't.
